Sources/Stagehand/AnimationCurve/AnimationCurve+Spring.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  Copyright 2025 Block Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public struct SpringAnimationCurve: AnimationCurve {
+
+    /// - parameter damping: The damping ratio applied to the spring. Values should be in the range `[0,1]` where `1` means no oscillation (just smooth deceleration).
+    /// - parameter initialVelocity: The velocity at the start of the animation.
+    /// - parameter naturalFrequency: The natural frequency of the spring, which controls how "stiff" it feels. Higher values will cause the spring to bounce more times.
+    public init(damping: CGFloat, initialVelocity: CGFloat, naturalFrequency: CGFloat = 10) {
+        self.damping = damping
+        self.initialVelocity = initialVelocity
+        self.naturalFrequency = naturalFrequency
+    }
+
+    private let damping: CGFloat
+    private let initialVelocity: CGFloat
+    private let naturalFrequency: CGFloat
+
+    public func adjustedProgress(for progress: Double) -> Double {
+        // Since the curve always starts at `(0,0)` and ends at `(1,1)`, these values should always be the same. Early
+        // return with the appropriate values here to avoid extra work and potential for rounding error.
+        if progress == 0 {
+            return 0
+        } else if progress == 1 {
```

Comment:
Any chance `progress` exceeds 1?

Reply:
It shouldn't, but easy enough to guard against 👍